lichbot.lic

```diff
@@ -122,10 +123,87 @@ class Lichbot
     fput "whisper #{character} Supported commands are:"
     fput "whisper #{character}   (help) - this list"
     fput "whisper #{character}   (inv|inventory) - view the items I have available"
+    fput "whisper #{character}   (bal|balance) - view your current bot balance"
+    fput "whisper #{character}   (with|withdraw) [amount] - withdraw [amount] from your bot balance."
     fput "whisper #{character}   (give|giveme) [item] - take an item from me"
     fput "whisper #{character}   (buffs|spells) - view the buffs I can cast"
     fput "whisper #{character}   (buff|cast) [abbreviation] - receive [abbreviation] buff from me"
-    fput "whisper #{character} I will also accept and stow any item you trade me"
+    fput "whisper #{character} I will also accept and stow any item you trade me, or accept coins via TIPs to hold for you."
+  end
+
+  def balance(character)
+    @last_action = Time.now
+    unless has_bot_account(character)
```

Comment:
Is there a need to manage whether or not they have an account? If you use `Hash.new(0)` you can default everyone to 0.